AirCastingTests/API/SyncContextServiceTests.swift

```diff
@@ -0,0 +1,175 @@
+                XCTAssertEqual(second["deleted"] as? Bool, false)
+                XCTAssertNil(second["version"])
+                exp.fulfill()
+            })
+            .store(in: &cancellables)
+        wait(for: [exp], timeout: 0.2)
+    }
+    
+    // MARK: - Error handling
+    
+    func test_whenServerReturnsError_finishesWithError() {
+        setupWithAPICallError(DummyError(errorData: "some_error"))
+        let exp = expectation(description: "Will finish with error")
+        service
+            .getSynchronizationContext(localSessions: .random)
+            .sink(receiveCompletion: { result in
+                defer { exp.fulfill() }
+                guard case .failure = result else { XCTFail("Expected to fail!"); return }
+            }, receiveValue: { _ in })
+            .store(in: &cancellables)
+        wait(for: [exp], timeout: 0.2)
+    }
+    
+    func test_whenServerReturnsMalformedData_finishesWithError() {
+        setupWithMalformedData()
+        let exp = expectation(description: "Will finish with error")
+        service
+            .getSynchronizationContext(localSessions: .random)
+            .sink(receiveCompletion: { result in
+                defer { exp.fulfill() }
+                guard case .failure = result else { XCTFail("Expected to fail!"); return }
+            }, receiveValue: { _ in })
+            .store(in: &cancellables)
+        wait(for: [exp], timeout: 0.2)
+    }
+    
+    // MARK: - Fixture setup
+    
+    private func setupWithCorrectDataReturned() {
+        client.requestTaskStub = { request, completion in
+            let response: (data: Data, response: HTTPURLResponse) = (data: self.validContextResponseData, response: .success())
+            // We need the client to be asynchronous, see details in implementation file.
+            DispatchQueue.global().async {
+                completion(.success(response), request)
+            }
+        }
+    }
+    
+    private func setupWithAPICallError(_ error: Error) {
+        client.requestTaskStub = { request, completion in
+            // We need the client to be asynchronous, see details in implementation file.
+            DispatchQueue.global().async {
+                completion(.failure(error), request)
+            }
+        }
+    }
+    
+    private func setupWithMalformedData() {
+        client.requestTaskStub = { request, completion in
+            // 1KB of zeros
+            let response: (data: Data, response: HTTPURLResponse) = (data: .init(count: 1024), response: .success())
+            DispatchQueue.global().async {
+                completion(.success(response), request)
+            }
+        }
+    }
+}
+
+extension Array where Element == SessionsSynchronization.Metadata {
+    static var random: [Element] = [.random, .random]
+}
+
+class RequestAuthorizationServiceMock: RequestAuthorisationService {
+    struct DummyError: Error {
+        
+    }
```

Comment:
It's empty, don't know if it should be.. wondering

Reply:
since `Error` is a protocol we need any implementation for it to use in test situations. So therefore the need for `DummyError`. But good you mention this since I also created a slightly less dummy DummyError as a universal test utility:
```Swift
struct DummyError: Error, Equatable {
    let errorData: String
    
    init(errorData: String = .default) {
        self.errorData = errorData
    }
}
```
So this one indeed can be removed!

Reply:
I didn't know that earlier, good to know. Thanks!